### ISSUE TITLE
TensorBoard integration for a hyper-parameter sampled by using `suggest_int(log=True)`

### ISSUE THREAD

**Issue by nzw0301** [optuna/optuna#2336]
## Motivation

The currently TensorBoard integration does not support `IntLogUniformDistribution` that is used for `suggest_int` with `log=True`.

## Description

Add lines for `IntLogUniformDistribution` [these lines](https://github.com/optuna/optuna/blob/master/optuna/integration/tensorboard.py#L60-L86):

```python
# this elif block
elif isinstance(param_distribution, optuna.distributions.IntLogUniformDistribution):
    self._hp_params[param_name] = hp.HParam(
        param_name, hp.IntInterval(param_distribution.low, param_distribution.high)
    )
...

distribution_list = [
    optuna.distributions.UniformDistribution.__name__,
    optuna.distributions.LogUniformDistribution.__name__,
    optuna.distributions.DiscreteUniformDistribution.__name__,
    optuna.distributions.IntUniformDistribution.__name__,
    optuna.distributions.IntLogUniformDistribution.__name__, # this line
    optuna.distributions.CategoricalDistribution.__name__,
]
```

I believe that adding test is also helpful.

**Comment by nzw0301** [comment]
I'll send a PR to fix this issue after PR #2335.

**Comment by nzw0301** [comment]
Ah, I realise that I should send after #2343 by @himkt.
